src/VectorStoreManager.ts

```diff
@@ -37,23 +33,11 @@ class VectorStoreManager {
   private saveDBDelay = 30000; // Save full DB every 30 seconds
   private hasUnsavedChanges = false;
 
-  constructor(
-    app: App,
-    settings: CopilotSettings,
-    encryptionService: EncryptionService,
-    getLangChainParams: () => LangChainParams
-  ) {
+  constructor(app: App) {
```

Comment:
no subscription to settings needed in this manager because all the methods always use the latest settings value